File: server/src/main/java/com/krillsson/sysapi/core/metrics/DriveMetrics.java
package com.krillsson.sysapi.core.metrics;

import com.krillsson.sysapi.core.domain.drives.Drive;
import com.krillsson.sysapi.core.domain.drives.DriveLoad;

import java.util.List;
import java.util.Optional;

public interface DriveMetrics {
    List<Drive> drives();

    List<DriveLoad> driveLoads();

    Optional<Drive> driveByName(String name);

    Optional<DriveLoad> driveLoadByName(String name);
}

File: server/src/main/java/com/krillsson/sysapi/core/metrics/DriveMetrics.kt
package com.krillsson.sysapi.core.metrics

import com.krillsson.sysapi.core.domain.drives.Drive
import com.krillsson.sysapi.core.domain.drives.DriveLoad
import java.util.*

interface DriveMetrics {
    fun drives(): List<Drive>
    fun driveLoads(): List<DriveLoad>
    fun driveByName(name: String): Optional<Drive>
    fun driveLoadByName(name: String): Optional<DriveLoad>
}